lib/tapioca/compilers/gem/events.rb

```diff
@@ -0,0 +1,122 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "pathname"
+
+module Tapioca
+  module Compilers
+    module Gem
+      class Event
+        extend T::Sig
+        extend T::Helpers
+
+        abstract!
+      end
+
+      class NewSymbolFound < Event
```

Comment:
I think I would prefer the "noun" - "verb" naming for these:
```
SymbolFound
ConstantFound
NodeAdded
ConstNodeAdded
ScopeNodeAdded
MethodNodeAdded
```

Reply:
Done in [91306f0](https://github.com/Shopify/tapioca/pull/785/commits/91306f0dcba1f49c616af03b6019520a36890d49) 👍